products/BellHybrid/services/db/agents/MeditationStatsAgent.cpp

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "MeditationStatsAgent.hpp"
+#include "db/MeditationStatsMessages.hpp"
+
+#include <module-sys/Service/include/Service/Service.hpp>
+#include <module-vfs/paths/include/purefs/filesystem_paths.hpp>
+#include <MeditationStatisticsDB.hpp>
+
+namespace
+{
+    using namespace service::db::meditation_stats;
+
+    std::vector<Entry> transformRecords(const std::vector<db::meditation_stats::TableRow> &records)
+    {
+        std::vector<Entry> ret;
+        std::transform(records.cbegin(), records.cend(), std::back_inserter(ret), [](const auto &record) {
+            return Entry(record.timestamp, record.duration);
+        });
+        return ret;
+    }
+} // namespace
+
+namespace service::db::agents
+{
+    MeditationStats::MeditationStats(sys::Service *parentService, const std::string dbName)
+        : DatabaseAgent(parentService), dbName{dbName}, db{getDbFilePath().c_str()}
+    {}
+
+    void MeditationStats::registerMessages()
+    {
+        using std::placeholders::_1;
+        parentService->connect(messages::Add({}), std::bind(&MeditationStats::handleAdd, this, _1));
```

Comment:
I would use lambda(as a default functor) instead of std::bind  but its stylistic  
parentService->connect(messages::Add({}), [this](auto *req) { return handleAdd(req); });

Reply:
Good idea! It copy pasted it ;)